skrub/_squashing_scaler.py

```diff
@@ -0,0 +1,341 @@
+        lower_quantile : float, default=0.25
+            Value of the lower quantile that will be used
+            to determine the scaling factor to rescale the data.
+        upper_quantile : float, default=0.75
+            Value of the upper quantile that will be used
+            to determine the scaling factor to rescale the data.
+
+        Notes
+        -----
+
+        The formula for the transform is
+
+        .. math::
+
+            \begin{align*}
+                a &:= \begin{cases}
+                    1/(q_{\beta} - q_{\alpha}) &, q_{\beta} \neq q_{\alpha} \\
+                    2/(q_1 - q_0) &, q_{\beta} = q_{\alpha} \text{ and } q_1 \neq q_0 \\
+                    0 &, \text{ otherwise.}
+                \end{cases} \\
+                z &:= a(x - q_{1/2}), \\
+                x_{\mathrm{out}} &:= \frac{z}{\sqrt{1 + (z/B)^2}},
+            \end{align*}
+
+        where
+
+        - :math:`x` is a value in the input column.
+        - :math:`q_{\gamma}` is the :math:`\gamma`-quantile of the finite values in X,
+        - :math:`B` is max_abs_value
+        - :math:`\alpha` is lower_quantile
+        - :math:`\beta` is upper_quantile.
+        - :math:`x_{\mathrm{out}}` is the result of the computation that will be returned.
+
+        References
+        ----------
+        This method has been introduced as the robust scaling and smooth clipping transform in
+        `Better by default: Strong pre-tuned MLPs and boosted trees on tabular data
+        <https://arxiv.org/abs/2407.04491>`_ by Holzmüller, Grinsztajn, Steinwart (2024).
+
+        Examples
+        --------
+        >>> import pandas as pd
+        >>> import numpy as np
+        >>> from skrub import SquashingScaler
+
+        >>> tfm = SquashingScaler()
+        >>> X = pd.Series([-np.inf, -100.0, 0.0, 8.0, 9.0, 10.0, 20.0, 100.0, np.inf, np.nan], name='improvement')
+        >>> tfm.fit_transform(X)
+        0   -3.000000
+        1   -2.871295
+        2   -0.789352
+        3   -0.090867
+        4    0.000000
+        5    0.090867
+        6    0.948683
+        7    2.820284
+        8    3.000000
+        9         NaN
+        Name: improvement, dtype: float32
+        """  # noqa: E501
+
+    def __init__(
+        self,
+        max_absolute_value=3.0,
+        lower_quantile=0.25,
+        upper_quantile=0.75,
+    ):
+        self.max_absolute_value = max_absolute_value
+        self.lower_quantile = lower_quantile
+        self.upper_quantile = upper_quantile
+
+    def fit(self, X, y=None):
+        """Fit the transformer and transform a column.
+
+        Parameters
+        ----------
+        X : Pandas or Polars dataframe or series
+            The column to transform.
+        y : None
+            Unused. Here for compatibility with scikit-learn.
+
+        Returns
+        -------
+        self: SquashingScaler
+            The current object.
+        """
+        self.tfm_ = SingleColumnSquashingScaler(
```

Comment:
maybe we could make the transformer private by renaming it to `self._transformer` and exposing its learned attributes directly by copying them to the outer object's dict with something like

```python
if isinstance(self._transformer, OnEachColumn):
    self.median_ = [t.median_ for t in self._transformer.transformers_.values()]
    self.scale_ = [t.median_ for t in self._transformer.transformers_.values()]
else:
    self.median_ = self._transformer.median_
    self.scale_ = self._transformer.scale_
```

Reply:
(in this case we could add a section like

```
    Attributes
    ----------
    median_ : float or list of float
        median of the training data

    scale_ : ...
```

just below the 'parameters' section in the docstring )